libs/telemetry/src/main/java/org/opensearch/telemetry/metrics/DefaultMetricsRegistry.java

```diff
@@ -39,6 +43,11 @@ public Histogram createHistogram(String name, String description, String unit) {
         return metricsTelemetry.createHistogram(name, description, unit);
     }
 
+    @Override
+    public Closeable createObservableGauge(String name, String description, String unit, Supplier<Double> valueProvider, Tags tags) {
```

Comment:
Thinking about naming, I think we should not use `observable` here - this is just a gauge with value supplier, nothing more:

```suggestion
    public Closeable createGauge(String name, String description, String unit, Supplier<Double> valueProvider, Tags tags) {
```